### PR TITLE
Enabled header_saveload feature

### LINKED ISSUE #86: [7] #1669
https://github.com/bitshares/bitshares-ui/issues/1669

Preliminary questions, when you have tradingview visible on an exchange: 
 - can you login there with a tradingview account to see your settings?
 - does it remember your drawings even without being logged in or saving a template?

If you save a template it should obviously be available on the site when loaded next time

### PR DESCRIPTION
**Pull request by VerevkinAlexander**
Closes #86 
I enabled header_saveload feature

image

### CODE REVIEW

**Comment by sschiessl-bcp** [comment]
The cloud save and load feature works, but displays many other layouts
image

Where do they come from?

**Comment by sschiessl-bcp** [comment]
When I save a chart I see this additional selector
image

But when I then reload the website, that additional selector is gone. Where does it come from?

**Comment by VerevkinAlexander** [comment]
> Where do they come from?

They come from other users those use public user id
https://github.com/blockchainprojects/bitshares-ui/pull/89/files#diff-2e5f570765c5384009838dde2aa05e2eR89
We can create own or add to the settings store possibility to add own exist account for each user
https://github.com/mmmy/css3demos/wiki/Saving-and-Loading-Charts#managing-saved-charts-access

**Comment by VerevkinAlexander** [comment]
> that additional selector is gone. Where does it come from?

It caused because we initialize tradeview widget each time when we mount component

**Comment by sschiessl-bcp** [comment]
> > Where do they come from?
> 
> They come from other users those use public user id
> https://github.com/blockchainprojects/bitshares-ui/pull/89/files#diff-2e5f570765c5384009838dde2aa05e2eR89
> We can create own or add to the settings store possibility to add own exist account for each user
> https://github.com/mmmy/css3demos/wiki/Saving-and-Loading-Charts#managing-saved-charts-access

First step I would like to a user to only see templates saved by himself

**Comment by sschiessl-bcp** [comment]
> > > Where do they come from?
> > 
> > 
> > They come from other users those use public user id
> > https://github.com/blockchainprojects/bitshares-ui/pull/89/files#diff-2e5f570765c5384009838dde2aa05e2eR89
> > We can create own or add to the settings store possibility to add own exist account for each user
> > https://github.com/mmmy/css3demos/wiki/Saving-and-Loading-Charts#managing-saved-charts-access
> 
> First step I would like to a user to only see templates saved by himself

Is this implemented?

**Comment by VerevkinAlexander** [comment]
Not yet

**Comment by VerevkinAlexander** [comment]
there are three ways to implement it: 
- save data to Local storage
- save data to own data storage https://github.com/mmmy/css3demos/wiki/Saving-and-Loading-Charts#using-high-level-saveload
- save data to tiny storage provided by http://saveload.tradingview.com (unsafe) https://github.com/mmmy/css3demos/wiki/Saving-and-Loading-Charts#using-demo-charts-and-study-templates-storage

**Comment by VerevkinAlexander** [comment]
Which way should I choose?

**Comment by VerevkinAlexander** [comment]
I`ve added save /load charts functionality please look.  I`ve temporarily used  in memory charts saving

**Comment by sschiessl-bcp** [comment]
LocalStorage would be preferable

**Comment by sschiessl-bcp** [comment]
Accidently closed

**Comment by VerevkinAlexander** [comment]
I`ve implemented in local storage saving and charts deleting. Please review it

**Comment by VerevkinAlexander** [comment]
I think what it will be cool if users have the possibility to share their charts layout. What do you think about it?

**Comment by sschiessl-bcp** [comment]
> I think what it will be cool if users have the possibility to share their charts layout. What do you think about it?

I agree, in what way would you propose this to be done?

**Comment by VerevkinAlexander** [comment]
We can add export via JSON file uploading also it can be coded in base64 format. I wonder UI will have 2 buttons (expor and import) on the Load charts layout modal 
image

File: app/components/Exchange/TradingViewPriceChart.jsx [hunk 1]
```diff
@@ -147,18 +176,95 @@ export default class TradingViewPriceChart extends React.Component {
...
-        if (!!this.tvWidget) return false;
-        if (!np.marketReady) return false;
-        return true;
+    shouldComponentUpdate(np, state) {
+        return (
+            state.showLoadModal !== this.state.showLoadModal ||
+            state.showSaveModal !== this.state.showSaveModal ||
+            np.chartHeight !== this.props.chartHeight ||
+            this.props.charts.size !== np.charts.size ||
+            this.tvWidget ||
```
Comment (sschiessl-bcp) [inline app/components/Exchange/TradingViewPriceChart.jsx]: Please double check the logic when refactoring and adjust the mistake here

**Review by sschiessl-bcp — CHANGES_REQUESTED** [review]

**Comment by sschiessl-bcp** [comment]
Please rebase

File: app/components/Exchange/TradingViewPriceChart.jsx (hunk 1 above)
Comment (VerevkinAlexander) [reply app/components/Exchange/TradingViewPriceChart.jsx]: sorry, fixed

**Comment by sschiessl-bcp** [comment]
Accepted and forwarded

**Comment by VerevkinAlexander** [comment]
Rebased to newest develop
Plz review it

**Comment by VerevkinAlexander** [comment]
fixed issues https://github.com/bitshares/bitshares-ui/pull/2804#issuecomment-506967445

File: app/components/Exchange/Exchange.jsx [hunk 2]
```diff
@@ -3072,7 +3075,7 @@ class Exchange extends React.Component {
                 <div
                     className={cnames(
                         centerContainerWidth > 1200 &&
-                        (verticalOrderBook || verticalOrderBook)
+                            (verticalOrderBook || verticalOrderBook)
```
Comment (sschiessl-bcp) [inline app/components/Exchange/Exchange.jsx]: Whole Exchange file is reformat only, please revert (rebase if necessary)

**Review by sschiessl-bcp — CHANGES_REQUESTED** [review]

File: app/components/Exchange/Exchange.jsx (hunk 2 above)
Comment (VerevkinAlexander) [reply app/components/Exchange/Exchange.jsx]: I`ve changed https://github.com/blockchainprojects/bitshares-ui/pull/89/files/9ac96a83c547065d6e303fcbe16c8306eabb39ee#diff-3aaa4fc021c75bd17e1e59adb802e56dL2753 
Other were changed by pre-commit hook

**Comment by sschiessl-bcp** [comment]
?

**Comment by VerevkinAlexander** [comment]
> ?

I`ve updated develop and looks like it merged automatically because it has been merged to the origin develop

**Pull request merged**